_proposals/pup-0001.rst

```diff
@@ -0,0 +1,207 @@
+
+    Alternatives
+    ============
+
+    What other designs have been considered? What is the impact of not doing this?
+
+    Unresolved Questions
+    ====================
+
+    What parts of the design are still TBD?
+
+
+Reaching a Decision
+===================
+
+How Long?
+^^^^^^^^^
+
+The initial announcement needs to identify a must-respond-by date and time. This should be no less
+than 10 calendar days from the initial announcement. Longer periods are also OK, and can be adjusted
+via the list based on feedback via the list. I think using calendar days is better than business
+days because the whole world treats calendar days the same. I suggest 10 because sometimes people go
+offline for a whole week. Longer than 10 days and I think the process could become too slow.
+
+
+Deciding Yes/No
+^^^^^^^^^^^^^^^
+
+* If an obvious consensus arises via voting, the proposal is ratified. This is a bit vague, but it
+  works for the Django community.
+* Discussion can occur to confirm if consensus is obvious or not.
+* Votes come from anyone, not just core devs
+* A -1 from a core dev is considered a veto vote and will block ratification until the vote is
+  recast.
+* Core devs are expected to consider and echo serious feedback from the community especially any -1
+  votes from community members.
+* Consensus is not only possible. If obvious consensus is not reached, then the core devs decide.
+
+
+Why Approve/Reject This Way?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* It treats community decision making (not just core dev decision making) as a first class principle
+* It works for the best communities we know of (Django, Python, Apache, etc)
+* It allows people to be as involved or not as they way
+* It allows serious concerns to not be dismissed (due to a -1 veto vote from core dev)
+
+
+Approval
+^^^^^^^^
+
+* The proposal index is updated moving it from an active section to an approved area. This is
+  similar to the index of PEPs for Python.
+* A post-approval announcement should be sent in the same thread as the initial announcement and
+  discussion, preferably with the link to the approved proposal visible via pulpproject.org
+
+
+Rejected or Abandoned Proposals
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* They are moved in the index from the active section to the rejected or abandoned sections.
+* The proposal text stays committed and is not removed.
+
+
+Is This Replacing Story Planning in Redmine?
+============================================
+
+No. This is for changes to process and should not be used to discuss the feature set or bugs of
+Pulp. Feature and bug discussion would continue to occur as-is (on pulp.plan.io). Two recent-ish
+examples of things which could have gone through this process include "Napoleon docstrings or not?",
+"To Merge forward or not?".
+
+This is different than the communities we are modeling our process after. They use it for feature
+planning also, but I propose that we don't do that currently.
```

Comment:
I think this is good for now but I could see us eventually using this system for things that are too big to be captured by a single user story/feature (e.g. "Support Ansible").